src/execution.rs

```diff
@@ -5,14 +5,17 @@
 // Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
 // ------------------------------------------------------------------------------------------------
 
-use std::collections::HashMap;
+mod variables;
```

Comment:
There are a lot of `pub` types in variables.rs, which means that either this needs to be `pub mod` (so that external callers can name the types), or the types in variables.rs should actually be `pub(crate)`.

How do you feel about either of the following to clean it up a bit:

- Make `variables` a proper top-level module (so a sibling file to execution.rs, with a `pub mod` statement in lib.rs).  Change `pub use` below into `use`, since callers will name the type `tree_sitter_graph::variables::Globals` instead of `tree_sitter_graph::execution::Globals`.

- If most of the content of variables.rs is really local to execution.rs, that tells me it doesn't deserve to be a separate file, and all of the variable-related should still live here in execution.rs.  You can use section comments like [this](https://github.com/github/stack-graphs/blob/e17b7fc0d9555de333d4b27a7efe95e84c50401d/stack-graphs/src/graph.rs#L68-L69) to organize the contents of the file more clearly.

Reply:
Good catch. I wasn't entirely sure how the visibility should work here.

Let me explain what I wanted to achieve first:

- Most of `variables` is internal to `execution`.
- I personally don't like working with large files. I find `execution` already too large as it is, but it makes sense that these things are together. That's why I wanted to factor out the `variables` related types into that separate module.
- `Globals` is a something that is part of the crate interface. It uses `VariableMap` internally, so it seemed convenient to define it in the same module.
- I thought the fact that `variables` is `mod variables` in `execution` essentially meant it was only visible inside `execution`.

In light of this, I'd rather not merge `variables` into `execution` (and for `lazy_execution` there would be even of that, making that file even larger), but I don't think it should be a top-level module either. Scoping it as `pub(crate)` would be possible, but the scope could be even smaller---if that can be expressed.

Also, I don't think I understand how `mod`/`pub(X) mod` of the module interacts with the `pub(X)` definitions inside the module.

Reply:
After reading up on Rust modules, I think my idea is achieved with `pub(super)` visibility. I've updated the code to use that wherever possible. Is this an acceptable solution, or do you still have concerns?

Reply:
I'm okay with that!